mlir/lib/Analysis/DataFlow/LivenessAnalysis.cpp

```diff
@@ -309,15 +310,29 @@ RunLivenessAnalysis::RunLivenessAnalysis(Operation *op) {
              << " has no liveness info (unreachable), mark dead";
       solver.getOrCreateState<Liveness>(result.value());
     }
+    SmallVector<Value> mustLiveValues;
+    if (auto loopOp = dyn_cast<LoopLikeOpInterface>(op)) {
```

Comment:
That seems very ad-hoc to me to look for this specific interface, seems like a workaround for something more important going on: likely that all entry-block arguments should be live, unless RegionBranchOpInterface could say they aren't.

Reply:
I think you haven't understood what I'm saying: I'd like to see a fix that does not use any interface and treat the program in the most conservative way.